### PR TITLE
DO_NOT_MERGE: Wip apply clang format

### PR DESCRIPTION
**Pull request by hjmjohnson**
ONLY 1/2 Done refactoring.

Looking for quick comments/suggestions.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Remote/RTK.remote.cmake [hunk 1]
```diff
@@ -42,5 +42,5 @@ itk_fetch_module(
   "Reconstruction Toolkit (RTK) https://www.openrtk.org/"
   MODULE_COMPLIANCE_LEVEL 3
   GIT_REPOSITORY https://github.com/RTKConsortium/RTK.git
-  GIT_TAG ff1967638805ff73d028fde036bfd81954923ccf
+  GIT_TAG 94d070b0101e4f0a5a7d0649a4559b8ee6d22d63
   )
```
Comment (N-Dekker) [inline Modules/Remote/RTK.remote.cmake:46]: Please consider putting a remote module update in a separate pull request. Otherwise, it's just not clear to me why it is part of a "apply clang format" refactoring. Then please explain!

**Comment by N-Dekker** [comment]
STYLE: One declaration per line for readability
    STYLE: One declaration per line for readability
    STYLE: One declaration per line for readability
    STYLE: One declaration per line for readability 

Please use more specific commit message headers. Ideally a different message header for each commit. Or, if they all do exactly the same thing, please squash them together.

File: Modules/Core/Common/include/itkBresenhamLine.hxx [hunk 2]
```diff
@@ -61,6 +40,12 @@ BresenhamLine<VDimension>::BuildLine(LType Direction, IdentifierType length) ->
   {
     LastIndex[i] = (IndexValueType)(length * Direction[i]);
   }
+
+  // Direction of increment. -1 or 1
+  IndexType m_OverflowIncrement;
+  // Increment for the error for each step. Two times the difference between
+  // start and end
+  IndexType m_IncrementError;
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkBresenhamLine.hxx:48]: This commit seems unrelated to the commit message "One declaration per line for readability".

**Comment by N-Dekker** [comment]
Please make a separate pull request for each type of refactoring. Especially when they involve hundreds or more lines of code.

"One declaration per line" is a different type of refactoring than moving variable declarations closer to their first use.

File: Modules/Remote/RTK.remote.cmake (hunk 1 above)
Comment (hjmjohnson) [reply Modules/Remote/RTK.remote.cmake:46]: Yes.  I plan do create a separate PR for updating remotes.  I'm working with the developers to get fixes in upstream that will make a clean build.

File: Modules/Core/Common/include/itkGaussianDerivativeOperator.hxx [hunk 3]
```diff
@@ -41,15 +41,11 @@ GaussianDerivativeOperator<TPixel, VDimension, TAllocator>::GenerateCoefficients
...
-  double norm;
+  double norm{ 1.0 };
   if (m_NormalizeAcrossScale && m_Order)
   {
     norm = std::pow(m_Variance, m_Order / 2.0);
   }
-  else
-  {
-    norm = 1.0;
-  }
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkGaussianDerivativeOperator.hxx:52]: Maybe just one line of code:

    double norm{ (m_NormalizeAcrossScale && m_Order) ? std::pow(m_Variance, m_Order / 2.0) : 1.0 };

File: Modules/Core/Common/include/itkGaussianOperator.hxx [hunk 4]
```diff
@@ -59,19 +53,21 @@ GaussianOperator<TPixel, VDimension, TAllocator>::GenerateCoefficients() -> Coef
     }
   }
   // Normalize the coefficients so that their sum is one.
-  for (it = coeff.begin(); it < coeff.end(); ++it)
+  for (typename CoefficientVector::iterator it = coeff.begin(); it < coeff.end(); ++it)
   {
     *it /= sum;
   }
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkGaussianOperator.hxx:59]: Maybe a range-based for loop:

    for (auto & coefficient: coeff)
    {
      coefficient /= sum;
    }

File: Modules/Core/Common/include/itkHexahedronCell.hxx [hunk 5]
```diff
@@ -366,14 +356,14 @@ HexahedronCell<TCellInterface>::EvaluatePosition(CoordRepType *            x,
...
 
     //  calculate newton functions
-    for (unsigned int i = 0; i < Self::PointDimension3D; ++i)
-    {
-      fcol[i] = rcol[i] = scol[i] = tcol[i] = 0.0;
-    }
+    double fcol[Self::PointDimension3D] = { 0.0 };
+    double rcol[Self::PointDimension3D] = { 0.0 };
+    double scol[Self::PointDimension3D] = { 0.0 };
+    double tcol[Self::PointDimension3D] = { 0.0 };
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkHexahedronCell.hxx:362]: Nowadays, I tend to just write `{}` (rather than ` = { 0.0 }`, but that's just a matter of taste:

    double tcol[Self::PointDimension3D]{};

File: Modules/Core/Common/include/itkNeighborhood.hxx [hunk 6]
```diff
@@ -70,12 +70,7 @@ template <typename TPixel, unsigned int VDimension, typename TContainer>
 void
 Neighborhood<TPixel, VDimension, TContainer>::SetRadius(const SizeValueType s)
 {
-  SizeType k;
-
-  for (DimensionValueType i = 0; i < VDimension; ++i)
-  {
-    k[i] = s;
-  }
+  SizeType k{ s };
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkNeighborhood.hxx:73]: I don't think that's correct. I mean: `SizeType k{ s }` is equivalent to `SizeType k{ { s, 0, 0 } }` in 3-D.  If you want to "fill" all elements:

    auto k = SizeType::Filled(s);

**Comment by hjmjohnson** [comment]
@N-Dekker Thank you for your comments. 

This work is getting too much out of hand to do with this approach.  I'm going to step back and work on a smaller set of incremental updates.   

I appreciate your feedback very much. It will guide my second attempt at this.

FYI:  
I plan to submit a 
1). clang-tidy readability-isolate-declaration update that has very little value on it's own, but it greatly facilitates the next steps.
2) Make a pass that does many very trivial changes (almost no benefit on it's own). i.e. like 18d683ec792bfa7b7a950f781cf9836a5ef64a4d

3) Make a pass through modules making more aggressive changes.
4) -- I may ask you to run some of your scripts like c77f9f8da0d12d791159d0f23cfd7a6097c99f37

File: Modules/Core/Common/include/itkNeighborhood.hxx (hunk 6 above)
Comment (hjmjohnson) [reply Modules/Core/Common/include/itkNeighborhood.hxx:73]: This is one of the mistakes that is making me step back and start anew.

File: Modules/Core/Common/include/itkGaussianOperator.hxx (hunk 4 above)
Comment (hjmjohnson) [reply Modules/Core/Common/include/itkGaussianOperator.hxx:59]: After this set of refactorings, there is a cling-tidy filter that will make this type of change automatically across the toolkit.

It is one of the primary motivations for making these intermediary refactorings.
